IdentityCore/tests/automation/MSIDTestConfiguration.m

```diff
@@ -0,0 +1,318 @@
+            {
+                [accounts addObject:account];
+            }
+        }
+        else if ([responseDict[@"Users"] isKindOfClass:[NSArray class]])
+        {
+            for (NSDictionary *accountDict in responseDict[@"Users"])
+            {
+                MSIDTestAccount *account = [[MSIDTestAccount alloc] initWithJSONResponse:accountDict];
+
+                if (account)
+                {
+                    [accounts addObject:account];
+                }
+            }
+        }
+
+        _accounts = accounts;
+    }
+
+    return self;
+}
+
+- (instancetype)initWithJSONResponseData:(NSData *)response
+{
+    id responseObj = [NSJSONSerialization JSONObjectWithData:response options:0 error:nil];
+
+    if (!responseObj)
+    {
+        return nil;
+    }
+
+    NSDictionary *responseDict = nil;
+
+    // TODO: fix this hack on the server side!
+    if ([responseObj isKindOfClass:[NSArray class]])
+    {
+        NSArray *responseArray = (NSArray *) responseObj;
+        responseDict = responseArray[0];
+    }
+    else if ([responseObj isKindOfClass:[NSDictionary class]])
+    {
+        responseDict = responseObj;
+    }
+
+    return [self initWithJSONDictionary:responseDict];
+}
+
+- (NSString *)redirectURIFromArray:(NSArray *)redirectUris
+{
+    for (NSString *uri in redirectUris)
+    {
+        if ([uri hasPrefix:@"x-msauth"])
+        {
+            return uri;
+        }
+    }
+
+    return redirectUris[0];
+}
+
+- (NSDictionary *)configParameters
+{
+    return [self configParametersForAccount:nil];
+}
+
+- (NSDictionary *)configParametersForAccount:(MSIDTestAccount *)account
+{
+    return @{@"authority" : self.authority,
+             @"client_id" : self.clientId,
+             @"redirect_uri" : self.redirectUri,
+             @"resource" : self.resource};
+}
+
+- (NSString *)authority
+{
+    NSString *authorityHost = _authorityHost;
+    NSString *authority = _authorityHost;
+
+    // TODO: lab is fixing this hack on the server side and should be returning the full authority or just always the host
+    if (account.homeTenantId || (![authorityHost containsString:@"common"]
+                                 && ![authorityHost containsString:@"adfs"]
+                                 && ![authorityHost containsString:account.targetTenantId]))
+    {
+        if (account.homeTenantId)
+        {
+            authority = [authorityHost stringByAppendingString:account.targetTenantId];
+        }
+        else
+        {
+            authority = [authorityHost stringByAppendingString:@"common"];
+        }
+    }
+
+    return authority;
+}
+
+- (NSDictionary *)configParametersWithAdditionalParams:(NSDictionary *)additionalParams
+{
+    NSMutableDictionary *configParams = [[self configParameters] mutableCopy];
+    [configParams addEntriesFromDictionary:additionalParams];
+    return configParams;
+}
+
+- (NSDictionary *)configParametersWithAdditionalParams:(NSDictionary *)additionalParams
```

Comment:
how is account here being used??

Reply:
Is imaginary :D you won't see it being used. 
But seriously, removed the method.